### PR TITLE
cc65: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

File: recipes/cc65/all/conanfile.py [hunk 1]
```diff
@@ -1,163 +1,150 @@
...
+
+    def generate(self):
+        if is_msvc(self):
+            tc = MSBuildToolchain(self)
+            tc.generate()
+        tc = AutotoolsToolchain(self)
+        tc.make_args += [
+            f"PREFIX=/",
+            f"datadir=/bin/share/cc65",
+            f"samplesdir=/samples",
```
Comment (danimtb) [inline recipes/cc65/all/conanfile.py:68]:
```suggestion
            "PREFIX=/",
            "datadir=/bin/share/cc65",
            "samplesdir=/samples",
```

File: recipes/cc65/all/conanfile.py [hunk 2]
```diff
@@ -1,163 +1,150 @@
...
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        if is_msvc(self):
+            copy(self, "*.exe",
+                 dst=os.path.join(self.package_folder, "bin"),
+                 src=os.path.join(self.source_folder, "bin"),
+                 keep_path=False)
+            for dir in ("asminc", "cfg", "include", "lib", "target"):
+                copy(self, "*",
+                     dst=os.path.join(self.package_folder, "bin", "share", "cc65", dir),
+                     src=os.path.join(self.source_folder, dir))
```
Comment (danimtb) [inline recipes/cc65/all/conanfile.py:124]:
```suggestion
            for folder in ("asminc", "cfg", "include", "lib", "target"):
                copy(self, "*",
                     dst=os.path.join(self.package_folder, "bin", "share", "cc65", folder),
                     src=os.path.join(self.source_folder, folder))
```

**Review by danimtb — COMMENTED** [review]
Linter suggestions

**Review by danimtb — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Comment by SpaceIm** [comment]
@valgur migration of msvc branch is highly suspicious. See zimg recipe for an example of how to inject MSBuild conan toolchain (yep, it's not automatic, it would require https://github.com/conan-io/conan/pull/12817 which didn't get lot of interest from conan devs).

**Comment by valgur** [comment]
@SpaceIm Thanks for mentioning. You're welcome to open a PR to address this.

**Pull request merged**